pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -362,4 +365,64 @@ var _ = Describe("AWSNodeTemplateController", func() {
 			Expect(nodeTemplate.Status.SecurityGroups).To(BeNil())
 		})
 	})
+	Context("AMI Status", func() {
+		BeforeEach(func() {
+			awsEnv.EC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{
+				Images: []*ec2.Image{
+					{
+						Name:         aws.String("test-ami-1"),
+						ImageId:      aws.String("ami-test1"),
+						CreationDate: aws.String(time.Now().Format(time.UnixDate)),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String("test-ami-1")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+					{
+						Name:         aws.String("test-ami-2"),
+						ImageId:      aws.String("ami-test2"),
+						CreationDate: aws.String(time.Now().Format(time.UnixDate)),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String("test-ami-2")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+					{
+						Name:         aws.String("test-ami-3"),
+						ImageId:      aws.String("ami-test3"),
+						CreationDate: aws.String(time.Now().Format(time.UnixDate)),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String("test-ami-3")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+				},
+			})
+		})
+		It("Should expect no errors when AMI selector is not in the AWSNodeTemplate", func() {
+			nodeTemplate.Spec.AMISelector = nil
+			ExpectApplied(ctx, env.Client, nodeTemplate)
+			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+			amiIDsInStatus := lo.Map(nodeTemplate.Status.AMIs, func(ami v1alpha1.AMI, _ int) string {
+				return ami.Name
+			})
+			sort.Strings(amiIDsInStatus)
```

Comment:
Any reason we need to sort these? Should just be able to add them to a set and check that they're all there, since the order shouldn't matter.

Reply:
The nodeTemplate.Status.AMIs is not always sorted the same ways. So this would make the check consistent.